include/os/freebsd/spl/sys/proc.h

```diff
@@ -90,6 +90,9 @@ do_thread_create(caddr_t stk, size_t stksize, void (*proc)(void *), void *arg,
 	return (td);
 }
 
+#define	thread_create_named(name, stk, stksize, proc, arg, len,	\
+    pp, state, pri) \
+	do_thread_create(stk, stksize, proc, arg, len, pp, state, pri)
```

Comment:
It would be very useful to see the name passed through to `do_thread_create` and formatted into the string, like `"solthread %s (%p)", name, proc);`. For `thread_create` below passing `#proc` as the name should be useful as well.